sap/rfc/strust.py

```diff
@@ -0,0 +1,130 @@
+"""SAP STRUST utilities"""
+
+from collections import namedtuple
+
+Identity = namedtuple('Identity', ['pse_context', 'pse_applic'])
+
+CLIENT_ANONYMOUS = 'client_anonymous'
+CLIENT_STANDART = 'client_standart'
+
+IDENTITY_MAPPING = {
+    CLIENT_ANONYMOUS: Identity('SSLC', 'ANONYM'),
+    CLIENT_STANDART: Identity('SSLC', 'DFAULT')
+}
+
+
+class UploadCertError(Exception):
+    """Base exception for errors of this tool"""
+
+    # pylint: disable=unnecessary-pass
+    pass
+
+
+class InvalidSSLStorage(UploadCertError):
+    """Invalid SSL storage errors"""
+
+    # pylint: disable=unnecessary-pass
+    pass
+
+
+class PutCertificateError(UploadCertError):
+    """Adding Certificate errors"""
+
+    # pylint: disable=unnecessary-pass
+    pass
+
+
+class SSLCertStorage:
+    """SAP STRUST representation"""
+
+    def __init__(self, connection, pse_context, pse_applic):
+        self._connection = connection
+        self.identity = {
+            'PSE_CONTEXT': pse_context,
+            'PSE_APPLIC': pse_applic
+        }
+
+    def __repr__(self):
+        return 'SSL Storage {PSE_CONTEXT}/{PSE_APPLIC}'.format(**self.identity)
+
+    def __str__(self):
+        return repr(self)
+
+    def sanitize(self):
+        """Checks if the storage is OK and if not, raise an exception
+           of the type InvalidSSLStorage.
+        """
+
+        stat = self._connection.call(
+            'SSFR_PSE_CHECK',
+            {'IS_STRUST_IDENTITY': self.identity}
+        )
+
+        ret = stat['ET_BAPIRET2']
+        if not ret:
+            raise InvalidSSLStorage(
+                'The {0} is broken: received no response from the server - check STRUST manually.'
```

Comment:
Can you please fix this string message? There is no format in place but the temlate contains {0}.